src/sqlfluff/rules/L069.py

```diff
@@ -29,147 +35,366 @@ class Rule_L069(BaseRule):
+                WhitespaceSegment(),
+            ]
+            post_edits: List[BaseSegment] = [
+                SymbolSegment(")", type="end_bracket"),
+            ]
+            for _type in later_types:
+                edits = pre_edits + [_type] + in_edits + edits + post_edits
+
+        fixes = [
+            LintFix.replace(
+                context.segment,
+                edits,
+            )
+        ]
+        return fixes
+
+    @staticmethod
+    def _shorthand_fix_list(
+        context: RuleContext, shorthand_arg_1: BaseSegment, shorthand_arg_2: BaseSegment
+    ) -> List[LintFix]:
+        """Generate list of fixes to convert CAST and CONVERT to ShortHandCast."""
+        edits = [
+            shorthand_arg_1,
+            SymbolSegment("::", type="casting_operator"),
+            shorthand_arg_2,
+        ]
+
+        fixes = [
+            LintFix.replace(
+                context.segment,
+                edits,
+            )
+        ]
+        return fixes
+
     def _eval(self, context: RuleContext) -> Optional[LintResult]:
-        """Use ``CAST`` instead of ``CONVERT`` or ``::``."""
+        """Enforce consistent type casting style."""
         # Config type hints
-        self.allow_non_standard_type_casting: bool
+        self.preferred_type_casting_style: str
 
         # Rule disabled for teradata.
         if context.dialect.name == "teradata":
             return None
 
-        if not self.allow_non_standard_type_casting:
-            # Limit scope to CONVERT and casting_operator(::)
-            # CONVERT
-            if (
-                context.segment.is_type("function")
-                and context.segment.get_child("function_name").raw_upper == "CONVERT"
-            ):
-                # Get the content of CONVERT
-                convert_content = (
-                    FunctionalContext(context)
-                    .segment.children(sp.is_type("bracketed"))
-                    .children(
-                        sp.and_(
-                            sp.not_(sp.is_meta()),
-                            sp.not_(
-                                sp.is_type(
-                                    "start_bracket",
-                                    "end_bracket",
-                                    "whitespace",
-                                    "newline",
-                                    "comma",
-                                )
-                            ),
+        # Construct segment type casting
+        current_type_casting_style = (
+            "cast"
+            if context.segment.is_type("function")
+            and context.segment.get_child("function_name").raw_upper == "CAST"
+            else "convert"
+            if context.segment.is_type("function")
+            and context.segment.get_child("function_name").raw_upper == "CONVERT"
+            else "shorthand"
+            if context.segment.is_type("cast_expression")
+            else None
+        )
```

Comment:
I find the multiple chained ternary `if / else` hard to read. What would you think of changing it to more traditional `if / else`, something like:
```
if context.segment.is_type("function")
            and context.segment.get_child("function_name").raw_upper == "CAST":
    current_type_casting_style = "cast"
elif context.segment.is_type("function")
            and context.segment.get_child("function_name").raw_upper == "CONVERT"
    current_type_casting_style = "convert"
etc.
```

Reply:
I only used it to reduce the code but if it is hard to read then traditional it is then.